sklearn/preprocessing/_csr_polynomial_expansion.pyx

```diff
@@ -128,9 +219,16 @@ def _csr_polynomial_expansion(cnp.ndarray[DATA_T, ndim=1] data,
                 for j_ptr in range(i_ptr + interaction_only, row_ends):
                     j = indices[j_ptr]
                     if degree == 2:
-                        col = _deg2_column(d, i, j, interaction_only)
-                        expanded_indices[expanded_index] = col
-                        expanded_data[expanded_index] = (
+                        if max(i, j) > MAX_SAFE_INDEX_DEG2:
```

Comment:
Does `d` need to be a part of this condition?

If `d = intmax32` and `i = 2`, then `d * i` will overflow and `i**2 + 3*i` will not.